### ISSUE TITLE
2024-01-26 and 2024-01-27 Meeting Agendas

### ISSUE THREAD

**Issue by simojo** [ReadyResearchers-2023-24/cis-600-f2023-610-s2024-senior-thesis-simojo#4]
## :robot: JJ

* I have this laser driver that only works for Arduino code. How should I go about getting it to interface with ROS? I would rather not use an intermediate Arduino platform (for saving weight).
* Pausing the gazebo physics in conjunction with the PX4 program--I need to get it synchronized--any tips?
* Discussing timeline and getting strict due dates for everything.

**Comment by simojo** [comment]
* Understanding how to explain policy gradients and being able to define all of the variables.
  * I have been using [OpenAI's Deep RL book](https://spinningup.openai.com/en/latest/spinningup/rl_intro3.html) as reference.

**Comment by simojo** [comment]
## :atom: Willey

* Understanding inertial tensors better than I do already
* Standard way of explaining rotational matrices and why they're important

**Comment by simojo** [comment]
* Plates printed, trying to simulate this with much difficulty
* JJ did a gazebo rangefinder project almost 10 years ago. She's sure that things have changed. She wants to understand the connection between the simulation and what will be on the robot.
* A Q function is used in **Q learning**, and, it takes $s$ and $a$ and, using the Q table, you plug in all of the values, and, for all possible states and actions, you calculate values for all possible states and actions.
* JJ: forget about writing right now. I should be able to catch up later. As a collective group, they wanted to provide students with a good time frame. It is important to get the work done and to get good experimental data. **In 2 weeks, if we're not making steady enough progress, we will need to come up with some kind of back tracking plan.** If I tweak the parameters with gradient descent, it should work relatively well.
* JJ says that this may not be the best approach.
* Dedicate the rest of january and february to getting this work done.
* Doing the discrete approach will be easier to implement. If I am going to go for the combination approach, JJ says implement it fully discretely first, and then see how it does. Then, add on the continuous part!!!
* It may be possible to do a combination of discretized and continuous spaces.

**Comment by simojo** [comment]
* My project can still be successful even if I did not train successfully. I can still train and get failing results. I'm demonstrating what I tried and what didn't work. -- something to keep in mind.

**Comment by simojo** [comment]
#### My next steps

* Getting the simulation environment up and running with the sensors and 3D parts working.
* Simulate laser sensors
* explore learning strategies and discretizing things
* By mid-february, we can hope that the base line simulation is working
* Keep documenting all work and regularly updating repository.
